openedx/core/djangoapps/user_authn/views/login.py

```diff
@@ -387,6 +389,35 @@ def finish_auth(request):
     })
 
 
+def enterprise_selection_page(request, user, next_url):
+    """
+    Updates redirect url to enterprise selection page if user is associated
+    with multiple enterprises otherwise return the next url.
+
+    param:
+      next_url(string): The URL to redirect to after multiple enterprise selection or in case
+      the selection page is bypassed e.g when dealing with direct enrolment urls.
+    """
+    redirect_url = next_url
+
+    response = get_enterprise_learner_data_from_api(user)
+    if response and len(response) > 1:
+        redirect_url = '/enterprise/select/active/?success_url=' + next_url
+
+        # Check to see if next url has an enterprise in it. In this case if user is associated with
+        # that enterprise, activate that enterprise and bypass the selection page.
+        if re.match(r'/enterprise/.*/course/.*/enroll', next_url):
```

Comment:
Replace `.*` with and `settings.COURSE_KEY_REGEX` and make sure the enterprise uuid regex is also correct.
```suggestion
        if re.match(r'/enterprise/.*/course/{}/enroll'.format(settings.COURSE_KEY_REGEX), next_url):
```